core/src/main/java/org/apache/iceberg/avro/TypeToSchema.java

```diff
@@ -187,6 +187,21 @@ public Schema map(Types.MapType map, Schema keySchema, Schema valueSchema) {
     return mapSchema;
   }
 
+  @Override
+  public Schema variant() {
+    String recordName = "r" + fieldIds.peek();
```

Comment:
This showed up in tests. It would be good to update this so when the root type is a variant, a generic name is used or an exception is thrown. I also looked at the code to see if it would be reasonable to use the root name, but that is only passed in when the conversion is a struct or schema, so it would be a bit of work to have a user-supplied name.

Reply:
You mean to change the following namesFunction so the user can provide the naming mapping, right? I can address in the following PR.

>   private final BiFunction<Integer, Types.StructType, String> namesFunction;

BTW: I also update for struct to use default `table` as the name when the id is null.  

>   public Schema struct(Types.StructType struct, List<Schema> fieldSchemas) {
>     Integer fieldId = fieldIds.peek();
>     String recordName = namesFunction.apply(fieldId, struct);
>     if (recordName == null) {
>       recordName = fieldId != null ? "r" + fieldId : "table";
>     }

Reply:
I'd prefer not to change the behavior for the root record. It is possible to set that and I don't want any issues from an unnecessary change. The Variant change look good.

Reply:
Let me revert it for now. 

Basically in RemoveIds we have the following to add "table" as the name. 

>   static Schema removeIds(org.apache.iceberg.Schema schema) {
>     return AvroSchemaVisitor.visit(
>         AvroSchemaUtil.convert(schema.asStruct(), "table"), new RemoveIds());
>   }